R/iterate.R

```diff
@@ -71,14 +71,24 @@
 #'   the iteration should terminate.
 #' @param max_reqs The maximum number of requests to perform. Use `Inf` to
 #'   perform all requests until `next_req()` returns `NULL`.
+#' @param on_error What should happen if a request fails?
+#'
+#'   * `"stop"`, the default: stop iterating with an error.
+#'   * `"return"`: stop iterating, returning all the successful responses so
+#'     far, as well as an error object for the failed request.
 #' @param progress Display a progress bar? Use `TRUE` to turn on a basic
 #'   progress bar, use a string to give it a name, or see [progress_bars] to
 #'   customise it in other ways.
 #' @param path Optionally, path to save the body of request. This should be
 #'   a glue string that uses `{i}` to distinguish different requests.
 #'   Useful for large responses because it avoids storing the response in
 #'   memory.
-#' @return A list of [response()]s.
+#' @return
+#' A list, at most length `max_reqs`, containing [response]s and possibly one
```

Comment:
```suggestion
#' A list, at most length `max_reqs`, containing [response()]s and possibly one
```

Reply:
I think that makes it harder to read the docs?